pkg/client_data/lib/publisher_api.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
+
+part 'publisher_api.g.dart';
+
+/// Request payload when creating a publisher.
+@JsonSerializable()
+class CreatePublisherRequest {
+  /// The OAuth2 `access_token` to be used with webmasters API to prove
+  /// ownership of the domain.
+  ///
+  /// This must:
+  ///  * be valid at the time of the request,
+  ///  * obtained from the OAuth2 flow used on the pub.dev website,
+  ///  * have the scope: `'https://www.googleapis.com/auth/webmasters.readonly'`.
+  final String accessToken;
+
+  // json_serializable boiler-plate
+  CreatePublisherRequest({@required this.accessToken});
+  factory CreatePublisherRequest.fromJson(Map<String, dynamic> json) =>
+      _$CreatePublisherRequestFromJson(json);
+  Map<String, dynamic> toJson() => _$CreatePublisherRequestToJson(this);
+}
+
+/// Request payload when updating a publisher.
+@JsonSerializable()
+class UpdatePublisherRequest {
+  /// A human readable description in markdown.
+  ///
+  /// If left as `null` this field will be ignored, and the description of the
+  /// publisher will not be changed.
+  final String description;
+
+  /// Email to be set as contact email for the publisher.
+  ///
+  /// If left as `null` this field will be ignored.
+  /// If changed, the change will not take effect until the confirmation email
+  /// have been confirmed by the user.
+  final String contact;
```

Comment:
should we leave it out for now?

Reply:
Or... rather... for now it can be changed only to one of the members...

Reply:
didn't we resolve that we could use the consent flow for this?

Note. when writing code we can throw a UnimplementedError, until the day we implement it..
It's better to do many small PRs in quick successesion :)